### PR TITLE
Add documentation about OIDC and Federated Credentials

### LINKED ISSUE #10: Request for docs clarifications / guidance
Context: I'm an experience library developer who knows exactly zero (on a good day) about the machinery of code-signing, hence my libraries are not currently signed; "if only there was a MSFT service that could help me with this?!"

So: I have (after some stumbles) successfully created a Trusted Signing certificate profile in Azure; next step:"now let's sign something" - find this supported GitHub Action, and see:

``` yaml
      - name: Sign files with Trusted Signing
        uses: azure/trusted-signing-action@v0.3.15
        with:
          azure-tenant-id: ${{ secrets.AZURE_TENANT_ID }}
          azure-client-id: ${{ secrets.AZURE_CLIENT_ID }}
          azure-client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
```

ELI5; if we assume that a Trusted Signing certificate profile exists in an Azure account accessible to the user, how do I get from *there* to *here*? What CLIENT_ID and CLIENT_SECRET is it asking me for, and where would I find and/or generate them? The docs right now assume a lot of information that is probably obvious when you've done it at least once, but which is unfamiliar if you're coming in cold. Unfortunately the "learn" docs stop at the point when a certificate profile exists, and just [waves vaguely](https://learn.microsoft.com/en-us/azure/trusted-signing/how-to-signing-integrations#use-other-signing-integrations-with-trusted-signing) in the direction of "oh, and you might want to use it with GitHub Actions", without saying how.

### PR DESCRIPTION
**Pull request by japarson**
Closes #10 #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,14 +54,15 @@ jobs:
...
+Please review [Common error codes and mitigations](https://learn.microsoft.com/azure/trusted-signing/faq#common-error-codes-and-mitigations) before filing an issue.
 
 ### Authentication
 Behind the scenes, the Action uses [DefaultAzureCredential](https://learn.microsoft.com/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet) as the primary method of authentication to Azure. The [EnvironmentCredential](https://learn.microsoft.com/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) variables are exposed as inputs and then set to Action-scoped environment variables. Each credential type supported by `DefaultAzureCredential` can be disabled using the Action inputs.
 
 > [!NOTE]
-> [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role is required to successfully sign by using Trusted Signing
+> [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role is required to successfully sign by using Trusted Signing
```
Comment (riverar) [inline README.md:63]: Nit
```suggestion
> [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role is required to successfully sign with Trusted Signing
```

File: docs/OIDC.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+1. Assign the [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role to your service principal.
+   - Open your Trusted Signing Account in the Azure portal.
+     - **Note:** You can assign the role from your Resource Group or Subscription if you have multiple Trusted Signing accounts.
+   - Navigate to the Access Control (IAM) tab.
+   - Click 'Add role assignment'.
+   - Select 'Trusted Signing Certificate Profile Signer'.
+   - Next.
+   - Assign access to your 'User, group, or service principal' or 'Managed identity'.
+   - Review + assign.
```
Comment (riverar) [inline docs/OIDC.md:16]: Nit:

These steps need to be executed in order, so an ordered list is needed.

Code:
```markdown
1. Top-level item 1
   1. Sub-item 1
   2. Sub-item 2
   3. Sub-item 3
2. Top-level item 2
```

Output:
1. Top-level item 1
   1. Sub-item 1
   2. Sub-item 2
   3. Sub-item 3
2. Top-level item 2

File: README.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+<p align="center">
```
Comment (riverar) [inline README.md:1]: Nit: `align` attribute on `p` [is obsolete](https://dev.w3.org/html5/spec-LC/obsolete.html).

```suggestion
<p style="text-align: center;">
```

**Review by riverar — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (japarson) [reply README.md]: This doesn't seem to work correctly:
image

Comment (riverar) [reply README.md]: Oops, GitHub doesn't appear to support this. Ignore.

**Review by riverar — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (japarson) [reply README.md:1]: No problem. Thank you for the review!

**Review by chabiss — APPROVED** [review]

**Pull request merged**